### PR TITLE
Adding Modal component.

### PR DESCRIPTION
**Pull request by bipin7yadav**

### CODE REVIEW

File: modal.css [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+.side-nav:hover{
+    background-color: turquoise;
+    padding: 5px 3rem 5px 3rem;
+    border-radius: 1rem;
+}
+
+
+.sidebar{
+    width: 16rem;
+    height: fi;
```
Comment (srishti-maurya) [inline modal.css]: you might give it a vaild value or remove the `height`, if this is not intentional

File: modal.css [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    padding:1px;
+    top: 0;
+}
+
+.side-nav{
+    text-decoration: none;
+}
+
+.side-nav:hover{
+    background-color: turquoise;
```
Comment (srishti-maurya) [inline modal.css:33]: you may use color variables to re-use them globally

**Review by srishti-maurya — APPROVED** [review]
you may avoid styling the tags directly instead you may use classes to style the tags like you have styled the `button`, `html`, `header` and `body` directly.
Also you may add a separate file for the JS

File: modal.css [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    display: flex;
+    gap: 5px;
+    position: relative;
+}
+
+header{
+    display: flex;
+    justify-content: space-between;
+    margin: 5px;
+    background-color: rgb(155, 173, 238);
```
Comment (himadri2110) [inline modal.css:21]: Consider using color variables here

File: modal.css [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    border-bottom-left-radius: 1rem;
+    border-bottom-right-radius: 1rem;
```
Comment (himadri2110) [inline modal.css:23]: You can use shorthand property for `border-radius`

File: modal.css [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+.side-nav{
+    text-decoration: none;
+}
+
+.side-nav:hover{
+    background-color: turquoise;
+    padding: 5px 3rem 5px 3rem;
+    border-radius: 1rem;
```
Comment (himadri2110) [inline modal.css:35]: Stay consistent with using units. Either use px or rems, not both

**Review by himadri2110 — APPROVED** [review]

**Pull request merged**